src/components/GraphiQL.js

```diff
@@ -475,7 +475,13 @@ export class GraphiQL extends React.Component {
 
   _fetchQuery(query, variables, operationName, cb) {
     const fetcher = this.props.fetcher;
-    const fetch = fetcher({ query, variables, operationName });
+    const parsedVariables =
+      variables && variables.trim() !== '' ? JSON.parse(variables) : null;
```

Comment:
I'd be okay with a breaking change and communicating this thoroughly with the next release. Another prop sounds a bit over-complicated.